crates/conjure_core/src/ast/domains.rs

```diff
@@ -295,6 +304,83 @@ impl Domain {
         }
         self
     }
+
+    // simplified domain intersection function. defined for integer domains of sets
+    // TODO: does not consider unbounded domains yet
+    pub fn intersect(&self, other: &Domain) -> Option<Domain> {
+        match (self, other) {
+            (Domain::DomainSet(_, x), Domain::DomainSet(_, y)) => Some(Domain::DomainSet(
+                SetAttr::None,
+                Box::new((*x).intersect(y)?),
+            )),
+            (Domain::IntDomain(_), Domain::IntDomain(_)) => {
+                let mut v: Vec<i32> = vec![];
+                if self.is_finite()? && other.is_finite()? {
+                    if let (Some(v1), Some(v2)) = (self.values_i32(), other.values_i32()) {
+                        for values1 in v1.iter() {
+                            for values2 in v2.iter() {
```

Comment:
- rename values1 to value
- instead of another nested for loop, check whether value is in other using contains.